fury/ui/context.py

```diff
@@ -0,0 +1,120 @@
+"""UI context module."""
+
+import numpy as np
+
+
+class UIContextClass:
+    """Manage global UI context."""
+
+    _instance = None
+
+    def __new__(cls):
+        """Handle instance creation for the UI context singleton.
+
+        Returns
+        -------
+        UIContextClass
+            The single, shared instance of `UIContextClass`.
+        """
+        if cls._instance is None:
+            cls._instance = super(UIContextClass, cls).__new__(cls)
+            cls._instance._initialized = False
+        return cls._instance
+
+    def __init__(self):
+        """Initialize the UIContext singleton."""
+        if self._initialized:
+            return
+        self._use_v2 = True
+        self._hot_ui = None
+        self._active_ui = None
+        self._canvas_size = np.zeros(2)
+        self._initialized = True
+
+    def set_hot_ui(self, element):
+        """Set the currently 'hot' UI element.
+
+        Parameters
+        ----------
+        element : UI or None
+            UI element that is currently 'hot', or `None` if no element is hot.
+        """
+        self._hot_ui = element
+
+    def get_hot_ui(self):
+        """Get the currently 'hot' UI element.
+
+        Returns
+        -------
+        UI or None
+            UI element that is currently 'hot', or `None` if no element is hot.
+        """
+        return self._hot_ui
```

Comment:
Why not create a property? using @property and why not make it public if we are simply accessing it?

Reply:
Thanks, I missed those, will update